### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Alqupaty**
Refactoring classes hierarchies to be more ready to add new requirements.

### CODE REVIEW

File: src/com/hishamfactory/Company.java [hunk 1]
```diff
@@ -4,16 +4,19 @@
 
 public class Company {
     private String name;
+    private String uuid;
```
Comment (HIS-MOHAMMED) [inline src/com/hishamfactory/Company.java:7]: after company object created don't have UUID,fix it.

File: src/com/hishamfactory/Identifiable.java [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package com.hishamfactory;
+
+@FunctionalInterface
+public interface Identifiable {
```
Comment (HIS-MOHAMMED) [inline src/com/hishamfactory/Identifiable.java:4]: This method not useful and why created?

**Review by HIS-MOHAMMED — CHANGES_REQUESTED** [review]

File: src/com/hishamfactory/Company.java (hunk 1 above)
Comment (Alqupaty) [reply src/com/hishamfactory/Company.java:7]: fixed it

**Review by HIS-MOHAMMED — APPROVED** [review]
Changes accepted.

**Pull request merged**